enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,773 @@
+Kubernetes 1.27 introduced the scheduling readiness feature[^1]. It
+allows users and controllers to set scheduling gates in the
+`podSpec.schedulingGates` list at admission time. Gated pods allow mutating some
+of the pod spec fields before the pod is scheduled. In particular, the
+scheduler's default `Enqueue` plugin holds the pods before entering the
+`ActiveQ` queue if a scheduling gate exists in the `podSpec.schedulingGates`
+list. Until gates are in the list, the pod will expose its status as
+`SchedulingGated`. The controller responsible for a given scheduling gate should
+implement the business logic so that the pod can be considered ready to be
+scheduled. KEP-3838[^2] introduced the possibility of mutating the node affinity
+of pods while they are in `schedulingGated` status (i.e., their
+`podSpec.schedulingGates` list has at least one gate).
+
+The multiarch-manager-operator will leverage the scheduling readiness feature
+and the mutable pod scheduling directives to update the node affinity of pods
+with the set of architectures they support. When a pod is submitted to the API
+server, a mutating admission webhook intercepts the pod and adds a scheduling
+gate to the pod spec.
+
+The `PodPlacementController` watches the pod resource. Suppose any pod has the
+scheduling gate set. In that case, the images in the pod spec are inspected, a
+subset of architectures that satisfy all containers in the pod is built, and the
+pod spec is patched with the additional node affinity constraints. If any
+existing node selector requirements are set with key `kubernetes.io/arch`,
+the `PodPlacementController` does not override the behavior previously
+pushed by other stakeholders, leaving it as complies with KEP-3838[^2].
+
+The multiarch-manager-operator will be available as part of the OperatorHub, as
+a secondary operator managed by the Operator Lifecycle Manager (OLM). It will be
+a cluster-scoped operator that only cluster admins can install and configure.
+
+Its development is expected to ship in the following phases:
+
+- Phase 1 - Technology Preview (TP):
+  - The pod placement controller will be supported in High Availability (HA)
+    mode
+  - The webhook will be a stateless component
+  - The pod placement controller will support namespace filtering to skip pods
+    that should not be mutated
+  - No support for customized network scenarios (restricted, proxied,
+    disconnected...)
+  - No TLS verification for the registries as we are only fetching metadata
+  - No caching facilities for the image metadata
+  - No graceful uninstallation support: any remaining gated pods have to be
+    manually patched to resume scheduling after uninstallation
+- Phase 2 - General Availability (GA):
+  - support for customized network scenarios (restricted, proxied,
+    disconnected...)
+  - support mirror registries through ICSPs/IDMSs
+  - implement caching facilities for the image metadata
+- Phase 3:
+  - TLS verification for the registries
```

Comment:
This must exist from the very start, not as a low priority item that is the first thing to cut.